RuntimePermissions/app/src/main/java/sample/github/nisrulz/runtimepermissions/MainActivity.java

```diff
@@ -89,6 +86,16 @@ public void onRequestPermissionsResult(int requestCode, String[] permissions, in
     private void makeACall() {
         Intent i = new Intent(Intent.ACTION_CALL);
         i.setData(Uri.parse("tel:90990990900"));
+        if (ActivityCompat.checkSelfPermission(this, Manifest.permission.CALL_PHONE) != PackageManager.PERMISSION_GRANTED) {
+            // TODO: Consider calling
+            //    ActivityCompat#requestPermissions
+            // here to request the missing permissions, and then overriding
+            //   public void onRequestPermissionsResult(int requestCode, String[] permissions,
+            //                                          int[] grantResults)
+            // to handle the case where the user grants the permission. See the documentation
+            // for ActivityCompat#requestPermissions for more details.
+            return;
+        }
```

Comment:
I understand your idea, but in code `makeACall()` function is called with a permission check everywhere. Your edit would be useful if the permission check was kept inside the method body itself and we remove the extra permission check from all other places. That way its required to be mentioned only inside the method body.